cpp/arcticdb/storage/azure/azure_mock_client.hpp

```diff
@@ -0,0 +1,60 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#pragma once
+
+#include <azure/core/http/curl_transport.hpp>
+#include <azure/core/http/http_status_code.hpp>
+#include <azure/core.hpp>
+#include <azure/storage/blobs.hpp>
+
+#include <arcticdb/storage/storage_utils.hpp>
+#include <arcticdb/storage/azure/azure_client_wrapper.hpp>
+
+namespace arcticdb::storage::azure {
+
+enum class AzureOperation{
+    READ,
+    WRITE,
+    DELETE, // delete_blobs will fail for all objects if one of them triggers a delete failure
+    LIST,
+    EXISTS,
+};
+
+class MockAzureClient : public AzureClientWrapper {
+
+private:
```

Comment:
Private at the bottom please

Reply:
Done in [ab07c4a](https://github.com/man-group/ArcticDB/pull/1331/commits/ab07c4a088ffabddde3a0c43103586332957ab1f)